Bug: `takt add` で存在するブランチが「Base branch does not exist」と誤判定される

## Summary

`gitRefExists()` が `git check-ref-format --branch -- <ref>` を実行しているが、`--branch` モードは引数を1つしか受け取らない。`--` を挟むことで引数が2つになり usage error（exit 129）となるため、`branchExists()` が常に `false` を返す。

結果として、`takt add` がローカル・リモート両方に存在するブランチに対して「Base branch does not exist」と誤判定する。

## Environment

- takt: 0.30.0
- Git: 2.53.0（macOS）で再現確認
  - `--branch` モードは仕様上つねに引数を1つだけ受け取るため、他バージョンでも同様の可能性あり（[git-check-ref-format Documentation](https://git-scm.com/docs/git-check-ref-format)）

## Affected File

`src/infra/task/clone-base-branch.ts` — `gitRefExists()`, `assertValidBranchRef()`

## Steps to Reproduce

```bash
$ git check-ref-format --branch "feature/xxx-branch"
feature/xxx-branch    # exit 0 — OK

$ git check-ref-format --branch -- "feature/xxx-branch"
usage: git check-ref-format [--normalize] [<options>] <refname>
   or: git check-ref-format --branch <branchname-shorthand>
                              # exit 129 — usage error
```

`takt add` 実行時の症状:

```
現在のブランチ: feature/xxx-branch
Base branch として feature/xxx-branch を使いますか？ [Y/n]: y
[ERROR] Base branch does not exist: feature/xxx-branch
Base branch (Enter for default):
```

## Expected

有効かつ存在するブランチ名に対して、`branchExists()` が `true` を返す。

## Actual

`git check-ref-format --branch -- <ref>` が usage error（exit 129）となり、`branchExists()` が常に `false` を返す。

## Root Cause

`--branch` モードは `<branchname-shorthand>` を1つだけ受け取る。`--` を挟むと引数が2つ（`--` と ref）になり usage error となる。

```typescript
// src/infra/task/clone-base-branch.ts
function gitRefExists(projectDir: string, ref: string): boolean {
  try {
    execFileSync('git', ['check-ref-format', '--branch', '--', ref], {
      //                                              ^^^^
      //  --branch モードに "--" と ref の2つの引数が渡されている
      cwd: projectDir,
      stdio: 'pipe',
    });
    execFileSync('git', ['rev-parse', '--verify', '--', ref], {
      cwd: projectDir,
      stdio: 'pipe',
    });
    return true;
  } catch {
    return false;  // ← check-ref-format の失敗でここに到達
  }
}

// 同様の問題が assertValidBranchRef() にもある
function assertValidBranchRef(projectDir: string, ref: string): void {
  try {
    execFileSync('git', ['check-ref-format', '--branch', '--', ref], {
      cwd: projectDir,
      stdio: 'pipe',
    });
  } catch {
    throw new Error(`Invalid base branch: ${ref}`);
  }
}
```

> **出典**: [git-check-ref-format Documentation](https://git-scm.com/docs/git-check-ref-format)
>
> ```
> SYNOPSIS
>     git check-ref-format [--normalize] [<options>] <refname>
>     git check-ref-format --branch <branchname-shorthand>
> ```
>
> `--branch` は独立したサブコマンド形式であり、`--` をオプション終端として受け付けない。

## Suggested Fix

`--branch` モードの引数から `--` を除去（2箇所）:

```diff
 // gitRefExists()
-    execFileSync('git', ['check-ref-format', '--branch', '--', ref], {
+    execFileSync('git', ['check-ref-format', '--branch', ref], {

 // assertValidBranchRef()
-    execFileSync('git', ['check-ref-format', '--branch', '--', ref], {
+    execFileSync('git', ['check-ref-format', '--branch', ref], {
```

## Impact

- `branchExists()` に依存する全機能が影響を受ける
  - `takt add` の base branch 検証（`worktree-settings.ts`）
  - `resolveBaseBranch()` のブランチ存在チェック

--- a/src/__tests__/clone-base-branch.test.ts
+++ b/src/__tests__/clone-base-branch.test.ts
@@ -1,0 +1,177 @@
+import { describe, it, expect, vi, beforeEach } from 'vitest';
+
+vi.mock('node:child_process', () => ({
+  execFileSync: vi.fn(),
+}));
+
+vi.mock('../shared/utils/index.js', async (importOriginal) => ({
+  ...(await importOriginal<Record<string, unknown>>()),
+  createLogger: () => ({
+    info: vi.fn(),
+    debug: vi.fn(),
+    error: vi.fn(),
+  }),
+}));
+
+vi.mock('../infra/config/index.js', () => ({
+  resolveConfigValue: vi.fn(),
+}));
+
+vi.mock('../infra/task/branchList.js', () => ({
+  detectDefaultBranch: vi.fn().mockReturnValue('main'),
+}));
+
+import { execFileSync } from 'node:child_process';
+import { resolveConfigValue } from '../infra/config/index.js';
+import { branchExists, resolveBaseBranch } from '../infra/task/clone-base-branch.js';
+
+const mockExecFileSync = vi.mocked(execFileSync);
+const mockResolveConfigValue = vi.mocked(resolveConfigValue);
+
+beforeEach(() => {
+  vi.clearAllMocks();
+});
+
+describe('branchExists — check-ref-format argument correctness', () => {
+  it('should call check-ref-format with --branch and ref only (no --)', () => {
+    // Given: both check-ref-format and rev-parse succeed
+    const checkRefFormatCalls: string[][] = [];
+
+    mockExecFileSync.mockImplementation((_cmd, args) => {
+      const argsArr = args as string[];
+      if (argsArr[0] === 'check-ref-format') {
+        checkRefFormatCalls.push([...argsArr]);
+        return Buffer.from('feature/my-branch\n');
+      }
+      if (argsArr[0] === 'rev-parse') {
+        return Buffer.from('abc123\n');
+      }
+      return Buffer.from('');
+    });
+
+    // When
+    branchExists('/project', 'feature/my-branch');
+
+    // Then: check-ref-format was called with exactly ['check-ref-format', '--branch', 'feature/my-branch']
+    expect(checkRefFormatCalls.length).toBeGreaterThanOrEqual(1);
+    expect(checkRefFormatCalls[0]).toEqual(['check-ref-format', '--branch', 'feature/my-branch']);
+  });
+
+  it('should not include -- in check-ref-format arguments for origin/ prefixed ref', () => {
+    // Given: local branch check fails, remote check succeeds
+    const checkRefFormatCalls: string[][] = [];
+    let callCount = 0;
+
+    mockExecFileSync.mockImplementation((_cmd, args) => {
+      const argsArr = args as string[];
+      if (argsArr[0] === 'check-ref-format') {
+        checkRefFormatCalls.push([...argsArr]);
+        return Buffer.from('');
+      }
+      if (argsArr[0] === 'rev-parse') {
+        callCount++;
+        // First rev-parse (local) fails, second (origin/) succeeds
+        if (callCount <= 1) {
+          throw new Error('branch not found');
+        }
+        return Buffer.from('abc123\n');
+      }
+      return Buffer.from('');
+    });
+
+    // When
+    branchExists('/project', 'feature/my-branch');
+
+    // Then: both check-ref-format calls should not contain '--'
+    expect(checkRefFormatCalls).toHaveLength(2);
+    expect(checkRefFormatCalls[0]).toEqual(['check-ref-format', '--branch', 'feature/my-branch']);
+    expect(checkRefFormatCalls[1]).toEqual(['check-ref-format', '--branch', 'origin/feature/my-branch']);
+  });
+});
+
+describe('resolveBaseBranch — assertValidBranchRef argument correctness', () => {
+  it('should call check-ref-format without -- when explicit baseBranch is provided', () => {
+    // Given: explicit baseBranch triggers assertValidBranchRef
+    const checkRefFormatCalls: string[][] = [];
+
+    mockResolveConfigValue.mockReturnValue(undefined);
+    mockExecFileSync.mockImplementation((_cmd, args) => {
+      const argsArr = args as string[];
+      if (argsArr[0] === 'check-ref-format') {
+        checkRefFormatCalls.push([...argsArr]);
+        return Buffer.from('');
+      }
+      if (argsArr[0] === 'rev-parse' && argsArr[1] === '--verify') {
+        return Buffer.from('abc123\n');
+      }
+      return Buffer.from('');
+    });
+
+    // When
+    resolveBaseBranch('/project', 'release/v2');
+
+    // Then: check-ref-format was called without '--'
+    expect(checkRefFormatCalls.length).toBeGreaterThanOrEqual(1);
+    const assertCall = checkRefFormatCalls[0]!;
+    expect(assertCall).toEqual(['check-ref-format', '--branch', 'release/v2']);
+  });
+
+  it('should throw Invalid base branch when check-ref-format fails for explicit baseBranch', () => {
+    // Given: check-ref-format throws for invalid ref
+    mockResolveConfigValue.mockReturnValue(undefined);
+    mockExecFileSync.mockImplementation((_cmd, args) => {
+      const argsArr = args as string[];
+      if (argsArr[0] === 'check-ref-format') {
+        throw new Error('invalid ref');
+      }
+      return Buffer.from('');
+    });
+
+    // When / Then
+    expect(() => resolveBaseBranch('/project', 'invalid..ref')).toThrow(
+      'Invalid base branch: invalid..ref',
+    );
+  });
+
+  it('should throw Base branch does not exist when branch does not exist', () => {
+    // Given: check-ref-format succeeds but rev-parse fails (branch not found)
+    mockResolveConfigValue.mockReturnValue(undefined);
+    mockExecFileSync.mockImplementation((_cmd, args) => {
+      const argsArr = args as string[];
+      if (argsArr[0] === 'check-ref-format') {
+        return Buffer.from('');
+      }
+      if (argsArr[0] === 'rev-parse') {
+        throw new Error('branch not found');
+      }
+      return Buffer.from('');
+    });
+
+    // When / Then
+    expect(() => resolveBaseBranch('/project', 'missing/branch')).toThrow(
+      'Base branch does not exist: missing/branch',
+    );
+  });
+
+  it('should not call assertValidBranchRef when no explicit baseBranch is provided', () => {
+    // Given: no explicit baseBranch, autoFetch off
+    mockResolveConfigValue.mockReturnValue(undefined);
+    const checkRefFormatCalls: string[][] = [];
+
+    mockExecFileSync.mockImplementation((_cmd, args) => {
+      const argsArr = args as string[];
+      if (argsArr[0] === 'check-ref-format') {
+        checkRefFormatCalls.push([...argsArr]);
+        return Buffer.from('');
+      }
+      return Buffer.from('');
+    });
+
+    // When
+    resolveBaseBranch('/project');
+
+    // Then: check-ref-format should not be called (no assertValidBranchRef)
+    expect(checkRefFormatCalls).toHaveLength(0);
+  });
+});
+

--- a/src/infra/task/clone-base-branch.ts
+++ b/src/infra/task/clone-base-branch.ts
@@ -7,7 +7,7 @@ const log = createLogger('clone');
 
 function gitRefExists(projectDir: string, ref: string): boolean {
   try {
-    execFileSync('git', ['check-ref-format', '--branch', '--', ref], {
+    execFileSync('git', ['check-ref-format', '--branch', ref], {
       cwd: projectDir,
       stdio: 'pipe',
     });
@@ -38,7 +38,7 @@ function resolveConfiguredBaseBranch(projectDir: string, explicitBaseBranch?: st
 
 function assertValidBranchRef(projectDir: string, ref: string): void {
   try {
-    execFileSync('git', ['check-ref-format', '--branch', '--', ref], {
+    execFileSync('git', ['check-ref-format', '--branch', ref], {
       cwd: projectDir,
       stdio: 'pipe',
     });

